metric-schema-java/src/integrationInputWithInject/java/com/palantir/test/ServerMetrics.java

```diff
@@ -0,0 +1,107 @@
+package com.palantir.test;
+
+import com.codahale.metrics.Gauge;
+import com.codahale.metrics.Histogram;
+import com.google.errorprone.annotations.CheckReturnValue;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.Safe;
+import com.palantir.tritium.metrics.registry.MetricName;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.util.Objects;
+import javax.inject.Inject;
+
+/**
+ * General web server metrics.
+ */
+public final class ServerMetrics {
+    private static final String LIBRARY_NAME = "witchcraft";
+
+    private static final String LIBRARY_VERSION =
+            Objects.requireNonNullElse(ServerMetrics.class.getPackage().getImplementationVersion(), "unknown");
+
+    private final TaggedMetricRegistry registry;
+
+    @Inject
+    public ServerMetrics(TaggedMetricRegistry registry) {
+        this.registry = registry;
```

Comment:
There are a few odd things going on here:
- new users will be fine, however future removal of jakarta-inject deps may result in compilation failures for folks who manually called the public constructor without realizing that's not recommended.
- The public constructor doesn't validate against nulls, resulting in potential NPEs after initialization

We scoped out a potential features that we haven't had the time to implement in which a namespace could declare tags across all nested metrics (much like library name/version, but user-defined). We'd end up with a builder for the metrics class e.g.

```java
ServerMetrics metrics = ServerMetrics.builder()
    .registry(taggedMetricRegistry)
    .serviceName(service)
    .endpointName(endpoint)
    .build();
Histogram responseSize = metrics.responseSize();
```
Such an improvement wouldn't work very well with ctor-based injection

Reply:
namespace scoped metrics would make this pretty painful.  If that's something that you think will get added I'd say just abandon this PR.  It's definitely not critical, just convenient.